src/main/java/jackyy/dimensionaledibles/block/BlockIslandCake.java

```diff
@@ -0,0 +1,238 @@
+package jackyy.dimensionaledibles.block;
+
+import com.feed_the_beast.ftblib.lib.data.FTBLibAPI;
+import jackyy.dimensionaledibles.DimensionalEdibles;
+import jackyy.dimensionaledibles.block.tile.TileIslandCake;
+import jackyy.dimensionaledibles.islands.Island;
+import jackyy.dimensionaledibles.islands.IslandManager;
+import jackyy.dimensionaledibles.registry.ModConfig;
+import mcjty.theoneprobe.api.ElementAlignment;
+import mcjty.theoneprobe.api.IProbeHitData;
+import mcjty.theoneprobe.api.IProbeInfo;
+import mcjty.theoneprobe.api.ProbeMode;
+import mcp.mobius.waila.api.IWailaConfigHandler;
+import mcp.mobius.waila.api.IWailaDataAccessor;
+import net.minecraft.block.ITileEntityProvider;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.init.Blocks;
+import net.minecraft.init.Items;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.*;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.TextFormatting;
+import net.minecraft.world.IBlockAccess;
+import net.minecraft.world.World;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.UUID;
+
+import static jackyy.dimensionaledibles.util.TeleporterHandler.teleport;
+import static jackyy.dimensionaledibles.util.TeleporterHandler.updateDimPos;
+
+public class BlockIslandCake extends BlockCakeBase implements ITileEntityProvider {
+
+
+    public BlockIslandCake() {
+        super();
+        setRegistryName(DimensionalEdibles.MODID + ":island_cake");
+        setTranslationKey(DimensionalEdibles.MODID + ".island_cake");
+    }
+
+    @Override
+    public void getDrops(NonNullList<ItemStack> drops, IBlockAccess world, BlockPos pos, IBlockState state, int fortune) {
+        TileIslandCake tile = (TileIslandCake) world.getTileEntity(pos);
+        if (tile != null) {
+
+            if (tile.isPersonalCake()) {
+                Item lockingItem = getItemFromResourceString(ModConfig.tweaks.islandCake.personalLockingItem);
+                if (lockingItem == null) {
+                    DimensionalEdibles.logger.warn("No locking item set, defaulting to minecraft:diamond");
+                    lockingItem = Items.DIAMOND;
```

Comment:
Hm, in this case if the locking item is null for some reason when breaking the cake, I just think that nothing should be added to the drops list